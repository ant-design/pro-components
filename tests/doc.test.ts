import demoTest from './demo';

demoTest('demos');
