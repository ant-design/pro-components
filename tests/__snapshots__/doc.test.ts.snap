// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`docs/components demos > üì∏ renders ./docs/components/customization-value-type.tsx correctly 1`] = `
<DocumentFragment>
  <div
    class="ant-app"
  >
    <div>
      test
    </div>
    <div
      class="ant-pro-card ant-pro-card-border"
      style="margin-block-end: 24px;"
    >
      <div
        class="ant-pro-card-header"
      >
        <div
          class="ant-pro-card-title"
        >
          SchemaForm
        </div>
      </div>
      <div
        class="ant-pro-card-body"
      >
        <form
          autocomplete="off"
          class="ant-form ant-form-vertical ant-pro-form"
          title="Ëá™ÂÆö‰πâ valueType"
        >
          <input
            style="display: none;"
            type="text"
          />
          <div
            class="ant-pro-form-group"
          >
            <div
              class="ant-pro-form-group-title"
            >
              Ê†áÁ≠æ
            </div>
            <div>
              <div
                class="ant-space ant-space-horizontal ant-space-align-start ant-pro-form-group-container "
                style="column-gap: 32px; row-gap: 0;"
              >
                <div
                  class="ant-space-item"
                >
                  <div
                    class="ant-form-item"
                  >
                    <div
                      class="ant-row ant-form-item-row"
                    >
                      <div
                        class="ant-col ant-form-item-label"
                      >
                        <label
                          class=""
                          for="name"
                          title="Âè™ËØªÈìæÊé•"
                        >
                          Âè™ËØªÈìæÊé•
                        </label>
                      </div>
                      <div
                        class="ant-col ant-form-item-control"
                      >
                        <div
                          class="ant-form-item-control-input"
                        >
                          <div
                            class="ant-form-item-control-input-content"
                          >
                            <a>
                              TradeCode 1
                            </a>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-space-item"
                >
                  <div
                    class="ant-form-item"
                  >
                    <div
                      class="ant-row ant-form-item-row"
                    >
                      <div
                        class="ant-col ant-form-item-label"
                      >
                        <label
                          class=""
                          for="name"
                          title="ÈìæÊé•"
                        >
                          ÈìæÊé•
                        </label>
                      </div>
                      <div
                        class="ant-col ant-form-item-control"
                      >
                        <div
                          class="ant-form-item-control-input"
                        >
                          <div
                            class="ant-form-item-control-input-content"
                          >
                            <input
                              class="ant-input ant-input-outlined"
                              id="name"
                              placeholder="ËØ∑ËæìÂÖ•ÈìæÊé•"
                              type="text"
                              value="TradeCode 1"
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-pro-form-group"
          >
            <div
              class="ant-pro-form-group-title"
            >
              Ë∑ØÂæÑ
            </div>
            <div>
              <div
                class="ant-space ant-space-horizontal ant-space-align-start ant-pro-form-group-container "
                style="column-gap: 32px; row-gap: 0;"
              >
                <div
                  class="ant-space-item"
                >
                  <div
                    class="ant-form-item"
                  >
                    <div
                      class="ant-row ant-form-item-row"
                    >
                      <div
                        class="ant-col ant-form-item-label"
                      >
                        <label
                          class=""
                          for="status"
                          title="Ê†áÁ≠æ"
                        >
                          Ê†áÁ≠æ
                        </label>
                      </div>
                      <div
                        class="ant-col ant-form-item-control"
                      >
                        <div
                          class="ant-form-item-control-input"
                        >
                          <div
                            class="ant-form-item-control-input-content"
                          >
                            <div
                              class="ant-space ant-space-horizontal ant-space-align-center ant-space-gap-row-small ant-space-gap-col-small"
                            >
                              <div
                                class="ant-space-item"
                              >
                                <span
                                  class="ant-tag"
                                >
                                  close
                                </span>
                              </div>
                              <div
                                class="ant-space-item"
                              >
                                <span
                                  class="ant-tag"
                                >
                                  close
                                </span>
                              </div>
                              <div
                                class="ant-space-item"
                              >
                                <input
                                  class="ant-input ant-input-sm ant-input-outlined"
                                  style="width: 78px;"
                                  type="text"
                                  value=""
                                />
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-space-item"
                >
                  <div
                    class="ant-form-item"
                  >
                    <div
                      class="ant-row ant-form-item-row"
                    >
                      <div
                        class="ant-col ant-form-item-label"
                      >
                        <label
                          class=""
                          for="status"
                          title="Âè™ËØªÊ†áÁ≠æ"
                        >
                          Âè™ËØªÊ†áÁ≠æ
                        </label>
                      </div>
                      <div
                        class="ant-col ant-form-item-control"
                      >
                        <div
                          class="ant-form-item-control-input"
                        >
                          <div
                            class="ant-form-item-control-input-content"
                          >
                            <span
                              class="ant-tag"
                            >
                              close
                            </span>
                            <span
                              class="ant-tag"
                            >
                              close
                            </span>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            style="display: flex; gap: 8px; align-items: center;"
          >
            <button
              class="ant-btn ant-btn-default ant-btn-color-default ant-btn-variant-outlined"
              type="button"
            >
              <span>
                Èáç ÁΩÆ
              </span>
            </button>
            <button
              class="ant-btn ant-btn-primary ant-btn-color-primary ant-btn-variant-solid"
              type="button"
            >
              <span>
                Êèê ‰∫§
              </span>
            </button>
          </div>
        </form>
      </div>
    </div>
    <div
      class="ant-pro-card ant-pro-card-border"
      style="margin-block-end: 24px;"
    >
      <div
        class="ant-pro-card-header"
      >
        <div
          class="ant-pro-card-title"
        >
          ProTable
        </div>
      </div>
      <div
        class="ant-pro-card-body"
      >
        <div
          class="ant-pro-table"
        >
          <div
            class="ant-pro-card ant-pro-table-search ant-pro-table-search-query-filter"
          >
            <form
              autocomplete="off"
              class="ant-form ant-form-horizontal ant-pro-query-filter ant-pro-form"
            >
              <input
                style="display: none;"
                type="text"
              />
              <div
                class="ant-row ant-row-start ant-pro-query-filter-row"
                style="margin-left: -12px; margin-right: -12px;"
              >
                <div
                  class="ant-col ant-col-8 ant-pro-query-filter-row-split"
                  style="padding-left: 12px; padding-right: 12px;"
                >
                  <div
                    class="ant-form-item"
                    style="flex-wrap: nowrap;"
                  >
                    <div
                      class="ant-row ant-form-item-row"
                    >
                      <div
                        class="ant-col ant-form-item-label"
                        style="flex: 0 0 80px;"
                      >
                        <label
                          class=""
                          for="name"
                          title="ÈìæÊé•"
                        >
                          ÈìæÊé•
                        </label>
                      </div>
                      <div
                        class="ant-col ant-form-item-control"
                        style="max-width: calc(100% - 80px);"
                      >
                        <div
                          class="ant-form-item-control-input"
                        >
                          <div
                            class="ant-form-item-control-input-content"
                          >
                            <input
                              class="ant-input ant-input-outlined"
                              id="name"
                              placeholder="ËØ∑ËæìÂÖ•ÈìæÊé•"
                              style="width: 100%;"
                              type="text"
                              value=""
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-col ant-col-8 ant-pro-query-filter-row-split"
                  style="padding-left: 12px; padding-right: 12px;"
                >
                  <div
                    class="ant-form-item"
                    style="flex-wrap: nowrap;"
                  >
                    <div
                      class="ant-row ant-form-item-row"
                    >
                      <div
                        class="ant-col ant-form-item-label"
                        style="flex: 0 0 80px;"
                      >
                        <label
                          class=""
                          for="status"
                          title="Ê†áÁ≠æ"
                        >
                          Ê†áÁ≠æ
                        </label>
                      </div>
                      <div
                        class="ant-col ant-form-item-control"
                        style="max-width: calc(100% - 80px);"
                      >
                        <div
                          class="ant-form-item-control-input"
                        >
                          <div
                            class="ant-form-item-control-input-content"
                          >
                            <div
                              class="ant-space ant-space-horizontal ant-space-align-center ant-space-gap-row-small ant-space-gap-col-small"
                            >
                              <div
                                class="ant-space-item"
                              >
                                <input
                                  class="ant-input ant-input-sm ant-input-outlined"
                                  style="width: 78px;"
                                  type="text"
                                  value=""
                                />
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-col ant-col-8"
                  style="padding-left: 12px; padding-right: 12px; text-align: end;"
                >
                  <div
                    class="ant-form-item ant-pro-query-filter-actions"
                  >
                    <div
                      class="ant-row ant-form-item-row"
                    >
                      <div
                        class="ant-col ant-form-item-label"
                      >
                        <label
                          class="ant-form-item-no-colon"
                          title=" "
                        >
                           
                        </label>
                      </div>
                      <div
                        class="ant-col ant-form-item-control"
                      >
                        <div
                          class="ant-form-item-control-input"
                        >
                          <div
                            class="ant-form-item-control-input-content"
                          >
                            <div
                              class="ant-space ant-space-horizontal ant-space-align-center"
                              style="column-gap: 16px; row-gap: 16px;"
                            >
                              <div
                                class="ant-space-item"
                              >
                                <div
                                  style="display: flex; gap: 8px; align-items: center;"
                                >
                                  <button
                                    class="ant-btn ant-btn-default ant-btn-color-default ant-btn-variant-outlined"
                                    type="button"
                                  >
                                    <span>
                                      Èáç ÁΩÆ
                                    </span>
                                  </button>
                                  <button
                                    class="ant-btn ant-btn-primary ant-btn-color-primary ant-btn-variant-solid"
                                    type="button"
                                  >
                                    <span>
                                      Êü• ËØ¢
                                    </span>
                                  </button>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </form>
          </div>
          <div
            class="ant-pro-card"
          >
            <div
              class="ant-pro-card-body"
              style="padding: 0px;"
            >
              <div
                class="ant-table-wrapper"
              >
                <div
                  class="ant-spin-nested-loading"
                >
                  <div
                    class="ant-spin-container"
                  >
                    <div
                      class="ant-table ant-table-middle"
                    >
                      <div
                        class="ant-table-container"
                      >
                        <div
                          class="ant-table-content"
                        >
                          <table
                            style="table-layout: auto;"
                          >
                            <colgroup />
                            <thead
                              class="ant-table-thead"
                            >
                              <tr>
                                <th
                                  class="ant-table-cell"
                                  scope="col"
                                >
                                  ÈìæÊé•
                                </th>
                                <th
                                  class="ant-table-cell"
                                  scope="col"
                                >
                                  Ê†áÁ≠æ
                                </th>
                              </tr>
                            </thead>
                            <tbody
                              class="ant-table-tbody"
                            >
                              <tr
                                class="ant-table-row ant-table-row-level-0"
                                data-row-key="1"
                              >
                                <td
                                  class="ant-table-cell"
                                >
                                  <a>
                                    TradeCode 1
                                  </a>
                                </td>
                                <td
                                  class="ant-table-cell"
                                >
                                  <span
                                    class="ant-tag"
                                  >
                                    close
                                  </span>
                                  <span
                                    class="ant-tag"
                                  >
                                    close
                                  </span>
                                </td>
                              </tr>
                            </tbody>
                          </table>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-pro-card ant-pro-card-border"
      style="margin-block-end: 24px;"
    >
      <div
        class="ant-pro-card-header"
      >
        <div
          class="ant-pro-card-title"
        >
          ProDescriptions
        </div>
      </div>
      <div
        class="ant-pro-card-body"
      >
        <div
          class="ant-descriptions ant-pro-descriptions"
        >
          <div
            class="ant-descriptions-view"
          >
            <table>
              <tbody>
                <tr
                  class="ant-descriptions-row"
                >
                  <td
                    class="ant-descriptions-item"
                    colspan="1"
                  >
                    <div
                      class="ant-descriptions-item-container"
                    >
                      <span
                        class="ant-descriptions-item-label"
                      >
                        ÈìæÊé•
                      </span>
                      <span
                        class="ant-descriptions-item-content"
                        style="min-width: 0;"
                      >
                        <a>
                          TradeCode 1
                        </a>
                      </span>
                    </div>
                  </td>
                </tr>
                <tr
                  class="ant-descriptions-row"
                >
                  <td
                    class="ant-descriptions-item"
                    colspan="1"
                  >
                    <div
                      class="ant-descriptions-item-container"
                    >
                      <span
                        class="ant-descriptions-item-label"
                      >
                        Ê†áÁ≠æ
                      </span>
                      <span
                        class="ant-descriptions-item-content"
                        style="min-width: 0;"
                      >
                        <span
                          class="ant-tag"
                        >
                          close
                        </span>
                        <span
                          class="ant-tag"
                        >
                          close
                        </span>
                      </span>
                    </div>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`docs/components demos > üì∏ renders ./docs/components/valueEnum-map.tsx correctly 1`] = `
<DocumentFragment>
  <div
    class="ant-app"
  >
    <div>
      test
    </div>
    <div
      class="ant-result ant-result-error"
    >
      <div
        class="ant-result-icon"
      >
        <span
          aria-label="close-circle"
          class="anticon anticon-close-circle"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="close-circle"
            fill="currentColor"
            fill-rule="evenodd"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M512 64c247.4 0 448 200.6 448 448S759.4 960 512 960 64 759.4 64 512 264.6 64 512 64zm127.98 274.82h-.04l-.08.06L512 466.75 384.14 338.88c-.04-.05-.06-.06-.08-.06a.12.12 0 00-.07 0c-.03 0-.05.01-.09.05l-45.02 45.02a.2.2 0 00-.05.09.12.12 0 000 .07v.02a.27.27 0 00.06.06L466.75 512 338.88 639.86c-.05.04-.06.06-.06.08a.12.12 0 000 .07c0 .03.01.05.05.09l45.02 45.02a.2.2 0 00.09.05.12.12 0 00.07 0c.02 0 .04-.01.08-.05L512 557.25l127.86 127.87c.04.04.06.05.08.05a.12.12 0 00.07 0c.03 0 .05-.01.09-.05l45.02-45.02a.2.2 0 00.05-.09.12.12 0 000-.07v-.02a.27.27 0 00-.05-.06L557.25 512l127.87-127.86c.04-.04.05-.06.05-.08a.12.12 0 000-.07c0-.03-.01-.05-.05-.09l-45.02-45.02a.2.2 0 00-.09-.05.12.12 0 00-.07 0z"
            />
          </svg>
        </span>
      </div>
      <div
        class="ant-result-title"
      >
        Something went wrong.
      </div>
      <div
        class="ant-result-extra"
      >
        ':scope +.ant-select-item-option-selected:not(.ant-select-item-option-disabled))+.ant-select-item-option-selected:not(.ant-select-item-option-disabled' is not a valid selector
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`docs/components demos > üì∏ renders ./docs/components/valueType.tsx correctly 1`] = `
<DocumentFragment>
  <div
    class="ant-app"
  >
    <div>
      test
    </div>
    <div
      class="ant-form-item"
    >
      <div
        class="ant-row ant-form-item-row"
      >
        <div
          class="ant-col ant-form-item-label"
        >
          <label
            class=""
            title="valueType ÈÄâÊã©"
          >
            valueType ÈÄâÊã©
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <div
                class="ant-select ant-select-outlined ant-select-in-form-item ant-pro-filed-search-select ant-select-single ant-select-allow-clear ant-select-show-search"
                style="width: 200px;"
              >
                <div
                  class="ant-select-selector"
                >
                  <span
                    class="ant-select-selection-wrap"
                  >
                    <span
                      class="ant-select-selection-search"
                    >
                      <input
                        aria-autocomplete="list"
                        aria-controls="rc_select_TEST_OR_SSR_list"
                        aria-expanded="false"
                        aria-haspopup="listbox"
                        aria-owns="rc_select_TEST_OR_SSR_list"
                        autocomplete="off"
                        class="ant-select-selection-search-input"
                        id="rc_select_TEST_OR_SSR"
                        role="combobox"
                        type="search"
                        value=""
                      />
                    </span>
                    <span
                      class="ant-select-selection-item"
                      title="ÊñáÊú¨Ê°Ü"
                    >
                      ÊñáÊú¨Ê°Ü
                    </span>
                  </span>
                </div>
                <span
                  aria-hidden="true"
                  class="ant-select-clear"
                  style="user-select: none;"
                  unselectable="on"
                >
                  <span
                    aria-label="close-circle"
                    class="anticon anticon-close-circle"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="close-circle"
                      fill="currentColor"
                      fill-rule="evenodd"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M512 64c247.4 0 448 200.6 448 448S759.4 960 512 960 64 759.4 64 512 264.6 64 512 64zm127.98 274.82h-.04l-.08.06L512 466.75 384.14 338.88c-.04-.05-.06-.06-.08-.06a.12.12 0 00-.07 0c-.03 0-.05.01-.09.05l-45.02 45.02a.2.2 0 00-.05.09.12.12 0 000 .07v.02a.27.27 0 00.06.06L466.75 512 338.88 639.86c-.05.04-.06.06-.06.08a.12.12 0 000 .07c0 .03.01.05.05.09l45.02 45.02a.2.2 0 00.09.05.12.12 0 00.07 0c.02 0 .04-.01.08-.05L512 557.25l127.86 127.87c.04.04.06.05.08.05a.12.12 0 00.07 0c.03 0 .05-.01.09-.05l45.02-45.02a.2.2 0 00.05-.09.12.12 0 000-.07v-.02a.27.27 0 00-.05-.06L557.25 512l127.87-127.86c.04-.04.05-.06.05-.08a.12.12 0 000-.07c0-.03-.01-.05-.05-.09l-45.02-45.02a.2.2 0 00-.09-.05.12.12 0 00-.07 0z"
                      />
                    </svg>
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <form
      autocomplete="off"
      class="ant-form ant-form-vertical ant-pro-form"
    >
      <input
        style="display: none;"
        type="text"
      />
      <div
        class="ant-pro-form-group"
      >
        <div
          class="ant-pro-form-group-title"
        >
          ÂàÜÁªÑ
        </div>
        <div>
          <div
            class="ant-space ant-space-horizontal ant-space-align-start ant-pro-form-group-container "
            style="column-gap: 32px; row-gap: 0;"
          >
            <div
              class="ant-space-item"
            >
              <div
                class="ant-form-item"
              >
                <div
                  class="ant-row ant-form-item-row"
                >
                  <div
                    class="ant-col ant-form-item-label"
                  >
                    <label
                      class="ant-form-item-required"
                      for="text"
                      title="ÁºñËæëÂô®"
                    >
                      ÁºñËæëÂô®
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <span
                          class="ant-input-affix-wrapper ant-input-affix-wrapper-focused ant-input-outlined pro-field pro-field-m"
                        >
                          <input
                            class="ant-input"
                            id="text"
                            placeholder="ËØ∑ËæìÂÖ•"
                            type="text"
                            value="123456"
                          />
                          <span
                            class="ant-input-suffix"
                          >
                            <span
                              class="ant-input-clear-icon"
                              role="button"
                              tabindex="-1"
                            >
                              <span
                                aria-label="close-circle"
                                class="anticon anticon-close-circle"
                                role="img"
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="close-circle"
                                  fill="currentColor"
                                  fill-rule="evenodd"
                                  focusable="false"
                                  height="1em"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M512 64c247.4 0 448 200.6 448 448S759.4 960 512 960 64 759.4 64 512 264.6 64 512 64zm127.98 274.82h-.04l-.08.06L512 466.75 384.14 338.88c-.04-.05-.06-.06-.08-.06a.12.12 0 00-.07 0c-.03 0-.05.01-.09.05l-45.02 45.02a.2.2 0 00-.05.09.12.12 0 000 .07v.02a.27.27 0 00.06.06L466.75 512 338.88 639.86c-.05.04-.06.06-.06.08a.12.12 0 000 .07c0 .03.01.05.05.09l45.02 45.02a.2.2 0 00.09.05.12.12 0 00.07 0c.02 0 .04-.01.08-.05L512 557.25l127.86 127.87c.04.04.06.05.08.05a.12.12 0 00.07 0c.03 0 .05-.01.09-.05l45.02-45.02a.2.2 0 00.05-.09.12.12 0 000-.07v-.02a.27.27 0 00-.05-.06L557.25 512l127.87-127.86c.04-.04.05-.06.05-.08a.12.12 0 000-.07c0-.03-.01-.05-.05-.09l-45.02-45.02a.2.2 0 00-.09-.05.12.12 0 00-.07 0z"
                                  />
                                </svg>
                              </span>
                            </span>
                          </span>
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-space-item"
            >
              <div
                class="ant-form-item"
              >
                <div
                  class="ant-row ant-form-item-row"
                >
                  <div
                    class="ant-col ant-form-item-label"
                  >
                    <label
                      class=""
                      for="text"
                      title="Âè™ËØª"
                    >
                      Âè™ËØª
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        123456
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        style="display: flex; gap: 8px; align-items: center;"
      >
        <button
          class="ant-btn ant-btn-default ant-btn-color-default ant-btn-variant-outlined"
          type="button"
        >
          <span>
            Èáç ÁΩÆ
          </span>
        </button>
        <button
          class="ant-btn ant-btn-primary ant-btn-color-primary ant-btn-variant-solid"
          type="button"
        >
          <span>
            Êèê ‰∫§
          </span>
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
